src/app/dev/DevToys.Tools/SmartDetection/XmlDataTypeDetector.cs

```diff
@@ -0,0 +1,37 @@
+﻿using DevToys.Tools.Helpers;
+using Microsoft.Extensions.Logging;
+
+namespace DevToys.Tools.SmartDetection;
+
+[Export(typeof(IDataTypeDetector))]
+[DataTypeName(PredefinedCommonDataTypeNames.Xml, baseName: PredefinedCommonDataTypeNames.Text)]
+internal sealed partial class XmlDataTypeDetector : IDataTypeDetector
+{
+    private readonly ILogger _logger;
+
+    [ImportingConstructor]
+    public XmlDataTypeDetector()
+    {
+        _logger = this.Log();
+    }
+
+    public ValueTask<DataDetectionResult> TryDetectDataAsync(
+        object data,
+        DataDetectionResult? resultFromBaseDetector,
+        CancellationToken cancellationToken)
+    {
+        if (resultFromBaseDetector is not null
+            && resultFromBaseDetector.Data is string dataString)
+        {
+            if (XmlHelper.IsValid(dataString, _logger))
+            {
+                return ValueTask.FromResult(new DataDetectionResult(Success: true, Data: dataString));
+            }
+        }
+
+        return ValueTask.FromResult(DataDetectionResult.Unsuccessful);
+    }
+
+    [LoggerMessage(1, LogLevel.Error, "An unexpected exception happened while trying to detect some XML data.")]
```

Comment:
Since this is a logger message and not shown to the user, I think it's better to keep it as just an English debug message for us. @veler, do you have an opinion?

Reply:
Well, I agree with @lwillia that as long as the error isn't displayed to the user, it's better to keep it in English. Otherwise it just makes it harder for us to analyze logs that customers shares to us.

That said, regarding this specific error log, I do not see `LogError` being invoked anywhere. Failure is correctly handled in `IsValid`, so this `LogError` method seems useless here. We can remove this code.

Reply:
That is an excellent point. Removed!